### PR TITLE
feat: first steps porting log manager UAT

### PR DESCRIPTION
**Pull request by Nelsonochoam**
**Issue #, if available:**

**Description of changes:**
This was some work created by @jaiszeen  on https://github.com/aws-greengrass/aws-greengrass-log-manager/pull/129. That old PR is getting closed an we are going with this one.

### CODE REVIEW

File: uat/src/main/java/com.aws.greengrass/FileSteps.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        List<String> randomMessages = generateRandomMessages(10, 1024);
+        for (String messageBytes : randomMessages) {
+            addDataToFile(messageBytes, file.toPath());
+        }
+    }
+
+    private static List<String> generateRandomMessages(int n, int length) {
+        List<String> msgs = new ArrayList<>();
+        for (int i = 0; i < n; i++) {
+            // TODO: Improves this as this is not how the logger writes the logs
```
Comment (dgggIoT) [inline uat/src/main/java/com.aws.greengrass/FileSteps.java:84]: Agree with TODO here, it is better to write as the real logger does

**Review by dgggIoT — APPROVED** [review]
Approve with some minor questions.

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@LogManager
+Feature: Greengrass V2 LogManager
+
+  As a customer, I want to selectively upload my logs and metrics to AWS Cloudwatch to save cost.
+
+  Background:
+    Given my device is registered as a Thing
+    And my device is running Greengrass
+    And 5 temporary rotated log files for component aws.greengrass.Nucleus
```
[the remark this answers is not in the document]
Comment (Nelsonochoam) [reply uat/src/main/resources/greengrass/features/log-manager-1.feature:9]: She is basically just porting what we have on EF here we can tinker and modify later.

**Review by saranyailla — APPROVED** [review]
Approving as we're copying everything over here from EF.

**Pull request merged**